tools/f18/f18.cc

```diff
@@ -478,6 +524,14 @@ int main(int argc, char *const argv[]) {
       driver.encoding = Fortran::parser::Encoding::UTF_8;
     } else if (arg == "-flatin") {
       driver.encoding = Fortran::parser::Encoding::LATIN_1;
+    } else if (arg == "-fget-definition") {
+      driver.getDefinition = true;
+      for (int i = 0; i < 3; i++) {
+        driver.sourcePosition[i] = std::stoi(args.front());
+        args.pop_front();
+      }
```

Comment:
What happens if the user provides incorrect input? (I.e. anything that isn't three numbers in the appropriate range.)

Reply:
Currently stoi throws an error if the argument provided is not an int, which happens when -fget-definitions is provided with less than 3 arguments. Should I catch the exception or just let it error?

Reply:
An unhandled exception is not a good way to report a user error. You can't catch the exception because we build with `-fno-exceptions`. Probably `std::strtol` is the best way to do this conversion.

Reply:
This comment remains unresolved.

Reply:
You might be looking at an outdated version. Github didn't mark this review as outdated.

Reply:
If this conversation is resolved, please click the "Resolve conversation" button.  The conversation appears on the current sources and is not marked as being "outdated".